syz-fuzzer/fuzzer.go

```diff
@@ -790,3 +794,30 @@ func kmemleakScan(report bool) {
 		panic(err)
 	}
 }
+
+// Checks if the KCOV device supports comparisons.
+func checkCompsEnabled() bool {
+	fd, err := syscall.Open("/sys/kernel/debug/kcov", syscall.O_RDWR, 0)
+	if err != nil {
+		return false
+	}
+	coverSize := uintptr(64 << 10)
+	_, _, errno := syscall.Syscall(
+		syscall.SYS_IOCTL, uintptr(fd), sys.KCOV_INIT_TRACE, coverSize)
+	if errno != 0 {
+		Logf(1, "KCOV_CHECK: KCOV_INIT_TRACE = %v", errno)
+		return false
+	}
+	_, err = syscall.Mmap(fd, 0, int(coverSize*8),
+		syscall.PROT_READ|syscall.PROT_WRITE, syscall.MAP_SHARED)
+	if err != nil {
+		Logf(1, "KCOV_CHECK: mmap = %v", err)
+		return false
+	}
+	KCOV_MODE_TRACE_CMP := uintptr(2)
```

Comment:
Eventually we should extract this value from kernel headers.
Add KCOV_MODE_TRACE_CMP to kcov_ioctls in sys.txt and then add:
```
# TODO: remove once the const is present in kernel headers.
define KCOV_MODE_TRACE_CMP 2
```
And then use sys.KCOV_MODE_TRACE_CMP here.